controllers/select_target.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/open-component-model/mpas-product-controller/api/v1alpha1"
+)
+
+// FilterTarget selects a target based on the provided filtering options.
+func (r *ProductDeploymentPipelineScheduler) FilterTarget(ctx context.Context, role v1alpha1.TargetRole, namespace string) (v1alpha1.Target, error) {
+	targetList := &v1alpha1.TargetList{}
+
+	m, err := v1.LabelSelectorAsSelector(&role.Selector)
+	if err != nil {
+		return v1alpha1.Target{}, fmt.Errorf("failed to parse label selectors to map: %w", err)
+	}
+
+	// We can't use client.MatchingFields for spec.type since we don't have a controller for Target
+	// and thus, we can't index the type field.
+	if err := r.List(ctx, targetList, client.InNamespace(namespace), client.MatchingLabelsSelector{
```

Comment:
Is that a sign that we should have a controller? Perhaps we should add a TODO here for post MVP.
```suggestion
	// We can't use client.MatchingFields for spec.type since we don't have a controller for Target
	// and thus, we can't index the type field.
	//TODO: post MVP add controller for targets
	if err := r.List(ctx, targetList, client.InNamespace(namespace), client.MatchingLabelsSelector{
```